src/Raven.Server/Documents/Handlers/CountersHandler.cs

```diff
@@ -754,6 +759,119 @@ private static void ThrowMissingDocument(string docId)
         }
+
+        public class ExecuteFixCounterGroupsCommand : TransactionOperationsMerger.MergedTransactionCommand
+        {
+            private readonly string _docId;
+            private readonly DocumentDatabase _database;
+
+            public ExecuteFixCounterGroupsCommand(DocumentDatabase database, string docId)
+            {
+                _docId = docId;
+                _database = database;
+            }
+
+            protected override long ExecuteCmd(DocumentsOperationContext context)
+            {
+                var numOfCounterGroupFixed = _database.DocumentsStorage.CountersStorage.FixCountersForDocument(context, _docId);
+                return numOfCounterGroupFixed;
+            }
+
+            public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto<TTransaction>(TransactionOperationContext<TTransaction> context)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [RavenAction("/databases/*/counters/fix", "PATCH", AuthorizationStatus.ValidUser, EndpointType.Write)]
+        public async Task FixCounterGroups()
+        {
+            var first = GetBoolValueQueryString("first", required: false) ?? true;
+            var task = FixAllCounterGroups();
+
+            if (first == false)
+            {
+                await task;
+                await NoContent();
+                return;
+            }
+
+            var tasks = new List<Task> { task };
+            var toDispose = new List<IDisposable>();
+            try
+            {
+                using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    foreach (var node in ServerStore.GetClusterTopology(context).AllNodes)
+                    {
+                        if (node.Value == Server.WebUrl)
+                            continue;
+
+                        var requestExecutor = ClusterRequestExecutor.CreateForShortTermUse(node.Value, Server.Certificate.Certificate, DocumentConventions.DefaultForServer);
+                        toDispose.Add(requestExecutor);
+
+                        var cmd = new FixCounterGroupsCommand(Database.Name);
+                        tasks.Add(requestExecutor.ExecuteAsync(cmd, context));
+                    }
+                }
+
+                await Task.WhenAll(tasks);
+                await NoContent();
+            }
+            finally
+            {
+                foreach (var disposable in toDispose)
+                {
+                    disposable.Dispose();
+                }
+            }
+        }
+
+        private async Task FixAllCounterGroups()
+        {
+            using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
```

Comment:
I would prefer to avoid keeping the read tx so long... instead, let's get 1024 docs at a time.